MoEngagePluginCards/MoEngagePluginCardsBridge.swift

```diff
@@ -0,0 +1,422 @@
+            let result = MoEngagePluginCardsUtil.buildHybridPayload(
+                forIdentifier: identifier,
+                containingData: [
+                    MoEngagePluginCardsContants.unClickedCardsCount: count
+                ]
+            )
+            completionHandler(result)
+        }
+    }
+}
+
+protocol MoEngagePluginCardsHandler {
+    func onAppOpenSync(
+        forAppID appID: String?,
+        withCompletion completionBlock: ((_ data: MoEngageCardSyncCompleteData?) -> Void)?
+    )
+
+    func onCardSectionLoaded(
+        forAppID appID: String?,
+        withCompletion completionBlock: ((_ data: MoEngageCardSyncCompleteData?) -> Void)?
+    )
+
+    func refreshCards(
+        forAppID appID: String?,
+        withCompletion completionBlock: ((_ data: MoEngageCardSyncCompleteData?) -> Void)?
+    )
+
+    func getCardsData(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock: @escaping ((_ cardsData: MoEngageCardsData?, _ accountMeta: MoEngageAccountMeta?) -> Void)
+    )
+
+    func getCardsCategories(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock:
+        @escaping ((_ categories: [String], _ accountMeta: MoEngageAccountMeta?) -> ())
+    )
+
+    func getCards(
+        forCategory category: String,
+        forAppID appID: String?,
+        withCompletionBlock completionBlock:
+        @escaping ((_ cards: [MoEngageCardCampaign], _ accountMeta: MoEngageAccountMeta?) -> Void)
+    )
+
+    func isAllCategoryEnabled(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock: @escaping ((Bool) -> Void)
+    )
+
+    func cardShown(_ card: MoEngageCardCampaign, forAppID appID: String?)
+
+    func cardClicked(_ card: MoEngageCardCampaign, forAppID appID: String?)
+    func cardClicked(_ card: MoEngageCardCampaign, withWidgetID widgetID: Int, forAppID appID: String?)
+
+    func cardDelivered(forAppID appID: String?)
+
+    func deleteCards(
+        _ cardsArr: [MoEngageCardCampaign],
+        forAppID appID: String?,
+        andCompletionBlock completionBlock:
+        @escaping ((_ isDeleted: Bool, _ accountMeta: MoEngageAccountMeta?) -> ())
+    )
+
+    func cardsViewControllerDismissed(forAppID appID: String?)
+
+    func getNewCardsCount(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock:
+        @escaping ((_ count: Int, _ accountMeta: MoEngageAccountMeta?) -> Void)
+    )
+
+
+    func getUnclickedCardsCount(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock:
+        @escaping ((_ count: Int, _ accountMeta: MoEngageAccountMeta?) -> Void)
+    )
+
+    func getClickedCardsCount(
+        forAppID appID: String?,
+        withCompletionBlock completionBlock:
+        @escaping ((_ count: Int, _ accountMeta: MoEngageAccountMeta?) -> Void)
+    )
+}
+
+extension MoEngageSDKCards: MoEngagePluginCardsHandler {}
```

Comment:
@msoumya-engg-sdk is this protocol really required?.. u can call the native methods directly right? . 
In some places native methods are called directly

Reply:
@RakshithaAcharya this is kept to make TDD easier where we can mock the data without having to actually invoke SDK APIs. Currently since initialization of cards model objects isn't exposed to pluginbase hence it is not being used, but we should keep it for future use for better test coverage. What are your thoughts?